modules/oidcsso/src/main/java/org/wso2/carbon/tomcat/oidcsso/extension/oidc/OIDCSSOValve.java

```diff
@@ -0,0 +1,319 @@
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.wso2.carbon.tomcat.oidcsso.extension.oidc;
+
+import org.apache.catalina.authenticator.SingleSignOn;
+
+import org.apache.catalina.connector.Request;
+import org.apache.catalina.connector.Response;
+
+import org.apache.juli.logging.Log;
+import org.apache.juli.logging.LogFactory;
+import org.wso2.carbon.tomcat.oidcsso.extension.Constants;
+import org.wso2.carbon.tomcat.oidcsso.extension.agent.OIDCAgent;
+import org.wso2.carbon.tomcat.oidcsso.extension.bean.AuthenticationResponse;
+import org.wso2.carbon.tomcat.oidcsso.extension.bean.OIDCLoggedInSession;
+import org.wso2.carbon.tomcat.oidcsso.extension.bean.RequestParameters;
+import org.wso2.carbon.tomcat.oidcsso.extension.bean.TokenResponse;
+import org.wso2.carbon.tomcat.oidcsso.extension.bean.UserInfoResponse;
+import org.wso2.carbon.tomcat.oidcsso.extension.utils.OIDCConfiguration;
+import org.wso2.carbon.tomcat.oidcsso.extension.utils.OIDCConfigurationLoader;
+import org.wso2.carbon.tomcat.oidcsso.extension.utils.exception.AuthenticationRequestException;
+import org.wso2.carbon.tomcat.oidcsso.extension.utils.exception.AuthenticationResponseException;
+import org.wso2.carbon.tomcat.oidcsso.extension.utils.exception.LogoutException;
+import org.wso2.carbon.tomcat.oidcsso.extension.utils.exception.TokenException;
+import org.wso2.carbon.tomcat.oidcsso.extension.utils.exception.UserInfoException;
+
+import java.io.IOException;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import javax.servlet.ServletException;
+
+/**
+ * This class implements an OpenID Connect valve for tomcat, which performs OpenID Connect based single-sign-on (SSO)
+ * and single-logout (SLO) functions.
+ */
+public class OIDCSSOValve extends SingleSignOn {
+    private static final Log log = LogFactory.getLog(OIDCSSOValve.class);
+    private OIDCAgent oidcAgent = new OIDCAgent();
+    private StateStore stateStore = new InMemoryStateStore();
+
+    @Override
+    public void invoke(Request request, Response response) throws IOException, ServletException {
+
+        Optional<OIDCConfiguration> configuration = OIDCConfigurationLoader.getOIDCConfiguration(request.getContext());
+        OIDCConfiguration oidcContextConfiguration;
+        if (configuration.isPresent()) {
+            oidcContextConfiguration = configuration.get();
+            if (oidcContextConfiguration == null) {
+                getNext().invoke(request, response);
+                return;
+            }
+        } else {
+            getNext().invoke(request, response);
+            return;
+        }
+        if (!oidcContextConfiguration.isEnabled()) {
+            getNext().invoke(request, response);
+            return;
+        }
+//        checks whether the request is a initial request to start the oidc flow by sending authentication request
```

Comment:
It is better to remove extra spaces at the beginning of the comment and align it with the next line of code.

Ex:-
// Checks whether the request is a initial request to start the oidc flow by sending authentication request
//  to openid provider
if ((request.getRequestURI().endsWith(Constants.SIGN_IN)) && ((request.getSession(false) == null)